linode/manager
# feat: [M3-6724]: Linode Config & Interface endpoints, validation, and React Query queries

## Description 📝
Add and modify existing Linode Config & Interface endpoints, validation, and RQ queries.

## Major Changes 🔄
- New Linode Configs API methods added
- New Linode Configs types added, some existing types modified
- Linode Configs RQ queries added
- Changes in `linodes.schema.ts`, modification to `linodeInterfaceSchema`
- Adjustments & fixes in `vpcs.schema.ts`

## How to test 🧪
As in the previous couple of PRs, there is some temporary code in `App.tsx` to work with a RQ query.

This PR significantly modifies the long-existing `linodeInterfaceSchema`, as well as newer VPC-related schemas added in previous PRs.

If you'd like to test those schemas thoroughly against the API spec, the best way would be by creating a new local project and pulling in this branch's api-v4 and validation packages as dependencies. Your `package.json` file in your project would look something like:

![Screenshot 2023-07-22 at 12 33 40 AM](https://github.com/linode/manager/assets/114682940/aee42d93-2f78-46f8-96ed-42f105826512)

From there you'd have a file like `file.js`, import the schemas you want to test, and create mock payloads. As an example:

```
import { linodeInterfaceSchema } from "@linode/validation";

const interfacePayload = {
  purpose: "vpc",
  primary: true,
  subnet: 15,
}; 

try {
  console.log(linodeInterfaceSchema.validateSync([interfacePayload]));
  linodeInterfaceSchema.validateSync([interfacePayload]);
} catch (err) {
  console.log(err);
}
```

And then run a node command like `node .src/file.js` to execute the code.

Beyond this, please confirm that you can still create Linodes in Cloud -- there were modifications made to `linodeInterfaceSchema` so that impacts the Linode creation payload.
